server/chat.js

```diff
@@ -1933,6 +1934,27 @@ Chat.fitImage = async function (url, maxHeight = 300, maxWidth = 300) {
 	return [Math.round(width * ratio), Math.round(height * ratio)];
 };
 
+/**
+ * Notifies a targetUser that a user was blocked from reaching them due to a setting they have enabled.
+ * @param {'pm'|'challenge'} blocked
+ * @param {User} targetUser
+ * @param {User} user
+ */
+Chat.maybeNotifyBlocked = function (blocked, targetUser, user) {
+	const options = 'or change it in the <button name="openOptions" class="subtle">Options</button> menu in the upper right.';
+	if (blocked === 'pm') {
+		if (!targetUser.blockPMsNotified) {
+			targetUser.send(`|html|The user '${user.name}' attempted to PM you but was blocked. To enable PMs, use /unblockpms ${options}`);
```

Comment:
Err, this sends to Lobby and will do nothing if Lobby isn't open.

Reply:
You probably want a PM from `~`, but that'll definitely notify. Maybe add some hardcoding to make it not notify?

Reply:
getting a PM to tell you that youre blocking PMs is kind of annoying. is there a way I can just make this show up in the chat of all the rooms the user is in?

Reply:
@Zarel

Reply:
All the rooms? That feels even worse to me. Some sort of "notification" that doesn't notify would be nice...

Reply:
I use the single panel layout, and a PM just causes the 'Home' tab to light up and its annoying to have to switch tabs to get a warning about something I don't care about (I don't have 'friends maliciously changing settings on my computer'). The only logical place in the single panel UI to show a message is in the chat stream, which is why I suggested that. Ideally the message would be sent to only the room that you're "in" (or PM if the user is in no rooms/battles), but I don't think the server knows which room a particular user is in at any given time (which wouldn't even work, because users can switch rooms and miss the message).

Failing any better UI for notification, a specially cased PM from `~` is I guess one solution, but I'm not super chuffed about it

Reply:
Hm, a popup would save you the need to switch rooms.

Reply:
I really don't like popups for non-user-initiated actions, though.

Reply:
Specially cased PM might be the best solution here.

Reply:
I've got `|pm|~|` working, but I need a way to distinguish the 'notification' PM in some way to know not to notify for it (string matching on the message would work, but seems super kludge)

Reply:
I guess a new message type `/nonotify`?

Reply:
Done (depends on Zarel/Pokemon-Showdown-Client#1280 which would need to land first).